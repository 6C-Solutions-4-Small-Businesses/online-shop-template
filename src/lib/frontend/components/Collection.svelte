<script lang="ts">
    import {goto} from "$app/navigation";
    import type {OfferSummaryPresentation} from "$lib/frontend/presentations/OfferSummaryPresentation";
    import ProductCard from "$lib/frontend/components/ProductCard.svelte";
    import {t} from '$translations/index'
    import ArrowLeftIcon from '$lib/frontend/components/icons/ArrowLeftIcon.svelte'

    export let id: string;
    export let name: string;
    export let products: OfferSummaryPresentation[];

    async function goToProductsPage(): Promise<void> {
        await goto(`/category/${id}`, {
            state: {
                name
            }
        });
    }
</script>

<div class="w-full pt-6 pb-10">
    <div class="px-10 py-5 justify-between items-center flex">
        <div class="text-primary text-2xl">{name}</div>

        <div class="justify-end items-center gap-2.5 flex">
            <button class="text-secondary flex items-center capitalize"
                    on:click={goToProductsPage}>
                {$t('collection.showMore')}

                <ArrowLeftIcon classNames="ml-2"/>
            </button>

            <div class="px-2 py-1 rounded-sm justify-center items-center gap-2.5 flex"></div>
        </div>
    </div>

    <div class="w-full p-5 grid gap-4 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 xl:grid-cols-5 2xl:grid-cols-6 3xl:grid-cols-6">
        {#each products as product}
            <ProductCard
                    id={product.id}
                    name={product.name}
                    image={product.image}
                    regularPrice={product.price}
                    unit={product.unit}
                    isSoldByQuantities={product.isSoldByQuantities}
                    salePrice={product.promotion?.salePrice}
                    percentage={product.promotion?.percentage}
            />
        {/each}
    </div>
</div>