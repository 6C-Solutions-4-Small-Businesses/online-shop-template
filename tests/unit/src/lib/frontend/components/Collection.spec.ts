import Collection from '$lib/frontend/components/Collection.svelte'
import {render} from '@testing-library/svelte'
import {describe, expect} from 'vitest'

describe('Collection', () => {
    it('should have match snapshot', () => {
        const {container} = render(Collection, {
            id: 'collection-id',
            name: 'collection-name',
            products: [
                {
                    id: 'a-product-id-1',
                    name: 'a-product-1-name',
                    price: 100,
                    image: 'image-1',
                    description: 'a product description',
                    quantity: 5,
                    isSoldByQuantities: false,
                    unit: 'Centiliter',
                },
                {
                    id: 'a-product-id-2',
                    name: 'a-product-2-name',
                    price: 100,
                    image: 'image-2',
                    description: 'a product description',
                    quantity: 10,
                    isSoldByQuantities: true,
                    unit: 'Centiliter',
                }
            ]
        })

        expect(container).toMatchSnapshot()
    })
})
