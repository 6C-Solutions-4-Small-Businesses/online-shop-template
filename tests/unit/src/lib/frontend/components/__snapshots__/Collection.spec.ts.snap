// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Collection > should have match snapshot 1`] = `
<div>
  <div
    class="w-full pt-8 pb-8 bg-green-50"
  >
    <div
      class="w-full flex justify-center text-3xl collection-name not-italic font-light mb-5 svelte-bzuqro"
    >
      collection-name
    </div>
     
    <div
      class="w-full p-5 grid gap-4 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 xl:grid-cols-5 3xl:grid-cols-6"
    >
      <div
        class="product-box relative w-full h-[22rem] sm:h-[19.5rem] md:h-[16.1rem] xl:h-[17rem] bg-white shadow-2xl svelte-d6kc1z"
        role="none"
      >
        <div
          class="absolute flex w-12 h-12 top-0 text-white items-center invisible svelte-d6kc1z"
        >
          <svg
            height="100"
            viewBox="0 0 490 490"
            width="100"
            x="0px"
            xml:space="preserve"
            y="0px"
          >
            <polygon
              points="0,0 490,0 0,490"
              style="fill: #ff0000;"
            />
          </svg>
           
          <div
            class="absolute -rotate-45 origin-center top-1 right-2 w-full text-[1rem] text-center font-semibold"
          >
            -
            null
            %
          </div>
        </div>
         
        <img
          alt="a-product-1-name"
          class="product-image object-contain absolute h-full w-full top-0 p-4"
          src="image-1"
          width="160"
        />
         
        <div
          class="w-full absolute flex justify-end"
        >
          <div
            class="text-2xl  w-2/12 h-10 flex items-center justify-center text-red-500"
            role="region"
          >
             
            <div
              class="border border-red-500 w-full h-full flex justify-center items-center bg-red-100"
            >
              <svg
                height="1.2em"
                viewBox="0 0 24 24"
                width="1.2em"
              >
                <path
                  d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3M7 18c-1.1 0-2 .9-2 2s.9 2 2 2s2-.9 2-2s-.9-2-2-2m10 0c-1.1 0-2 .9-2 2s.9 2 2 2s2-.9 2-2s-.9-2-2-2m-9.8-3.2v-.1l.9-1.7h7.4c.7 0 1.4-.4 1.7-1l3.9-7l-1.7-1l-3.9 7h-7L4.3 2H1v2h2l3.6 7.6L5.2 14c-.1.3-.2.6-.2 1c0 1.1.9 2 2 2h12v-2H7.4c-.1 0-.2-.1-.2-.2"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
         
        <div
          class="absolute bottom-0 right-0 w-full flex justify-end svelte-d6kc1z"
        >
          <div
            class="filter backdrop-blur-sm bg-green-300/40 font-sans w-28 h-[2.5rem] justify-center flex items-center text-lg svelte-d6kc1z"
            role="region"
          >
             
            <div
              class="w-full flex flex-col svelte-d6kc1z"
            >
               
              <div
                class="w-full items-center justify-center flex text-xl font-semibold svelte-d6kc1z"
              >
                <span
                  class="block"
                >
                  $
                  1
                  <sup>
                    00
                  </sup>
                </span>
                 
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="product-box relative w-full h-[22rem] sm:h-[19.5rem] md:h-[16.1rem] xl:h-[17rem] bg-white shadow-2xl svelte-d6kc1z"
        role="none"
      >
        <div
          class="absolute flex w-12 h-12 top-0 text-white items-center invisible svelte-d6kc1z"
        >
          <svg
            height="100"
            viewBox="0 0 490 490"
            width="100"
            x="0px"
            xml:space="preserve"
            y="0px"
          >
            <polygon
              points="0,0 490,0 0,490"
              style="fill: #ff0000;"
            />
          </svg>
           
          <div
            class="absolute -rotate-45 origin-center top-1 right-2 w-full text-[1rem] text-center font-semibold"
          >
            -
            null
            %
          </div>
        </div>
         
        <img
          alt="a-product-2-name"
          class="product-image object-contain absolute h-full w-full top-0 p-4"
          src="image-2"
          width="160"
        />
         
        <div
          class="w-full absolute flex justify-end"
        >
          <div
            class="text-2xl  w-2/12 h-10 flex items-center justify-center text-red-500"
            role="region"
          >
             
            <div
              class="border border-red-500 w-full h-full flex justify-center items-center bg-red-100"
            >
              <svg
                height="1.2em"
                viewBox="0 0 24 24"
                width="1.2em"
              >
                <path
                  d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3M7 18c-1.1 0-2 .9-2 2s.9 2 2 2s2-.9 2-2s-.9-2-2-2m10 0c-1.1 0-2 .9-2 2s.9 2 2 2s2-.9 2-2s-.9-2-2-2m-9.8-3.2v-.1l.9-1.7h7.4c.7 0 1.4-.4 1.7-1l3.9-7l-1.7-1l-3.9 7h-7L4.3 2H1v2h2l3.6 7.6L5.2 14c-.1.3-.2.6-.2 1c0 1.1.9 2 2 2h12v-2H7.4c-.1 0-.2-.1-.2-.2"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
         
        <div
          class="absolute bottom-0 right-0 w-full flex justify-end svelte-d6kc1z"
        >
          <div
            class="filter backdrop-blur-sm bg-green-300/40 font-sans w-28 h-[2.5rem] justify-center flex items-center text-lg svelte-d6kc1z"
            role="region"
          >
             
            <div
              class="w-full flex flex-col svelte-d6kc1z"
            >
               
              <div
                class="w-full items-center justify-center flex text-xl font-semibold svelte-d6kc1z"
              >
                <span
                  class="block"
                >
                  $
                  1
                  <sup>
                    00
                  </sup>
                </span>
                 
                <span
                  class="text-sm"
                >
                  <sub>
                    / 
                    cl
                  </sub>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="w-full flex justify-center items-center text-xl collection-name not-italic font-light h-10 mt-5 svelte-bzuqro"
    >
      <button
        class="hover:text-green-500"
      >
        Afficher Plus
      </button>
    </div>
  </div>
</div>
`;
